managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3270,10 +3274,32 @@ public long getOffloadedSize() {
         return propertiesMap;
     }
 
+    @Override
+    public void setProperty(String key, String value) throws InterruptedException {
+        Map<String, String> map = new HashMap<>();
+        map.put(key, value);
+        updateProperties(map, false, false, null);
+    }
+
+    @Override
+    public void deleteProperty(String key) throws InterruptedException {
+        updateProperties(null, false, true, key);
+    }
+
     @Override
     public void setProperties(Map<String, String> properties) throws InterruptedException {
+        updateProperties(properties, true, false, null);
+    }
+
+    @Override
+    public void asyncSetProperties(Map<String, String> properties, final SetPropertiesCallback callback, Object ctx) {
+        asyncUpdateProperties(properties, true, false, null, callback, ctx);
+    }
+
+    private void updateProperties(Map<String, String> properties, boolean isOverwrite, boolean isDelete,
```

Comment:
How is this reporting a failure in the operation?

Reply:
Done.